types/gulp-eslint/index.d.ts

```diff
@@ -0,0 +1,47 @@
+// Type definitions for gulp-eslint 6.0
+// Project: https://github.com/adametry/gulp-eslint
+// Definitions by: Martin Badin <https://github.com/martin-badin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Transform, Writable } from 'stream';
+import { CLIEngine } from 'eslint';
+
+declare namespace gulpEslint {
+    interface Result {
+        errorCount: number;
+        filePath: string;
+        messages: string[];
+        warningCount: number;
+    }
+
+    interface Results extends Array<Result> {
+        errorCount: number;
+        warningCount: number;
+    }
+
+    type Output = (() => Writable) | Writable;
+
+    type Formatter = (() => string) | string;
+
+    interface Plugin {
+        (options?: CLIEngine.Options | string): Transform;
+
+        result(action: (result: Result, callback: () => void) => void): Transform;
+        results(action: (results: Results, callback: () => void) => void): Transform;
+
+        failOnError(): Transform;
+        failAfterError(): Transform;
+
+        formatEach(formatter?: Formatter, output?: Output): Transform;
+        formatEach(output?: Output): Transform;
```

Comment:
Both can be omitted. Please see documentation (`Usage`).
https://github.com/adametry/gulp-eslint#usage

Reply:
@martin-badin Both can indeed, but not just the formatter. `formatEach(output?: Output)` allows you to specify a `Writer` as a first argument, which would not work, because the first argument is treated as a formatter.

Reply:
Thanks, @fasttime you are right. I updated PR.